pkg/controller/azdrivernode.go

```diff
@@ -101,10 +107,18 @@ func (r *ReconcileAzDriverNode) Recover(ctx context.Context) error {
 		if _, err = r.azClient.DiskV1beta2().AzDriverNodes(r.objectNamespace).Update(ctx, updated, metav1.UpdateOptions{}); err != nil {
 			return err
 		}
+		r.AddNodeInNodeDiskAvailabilityMap(ctx, node.Name, node.Labels)
 	}
 	return nil
 }
 
+func (r *ReconcileAzDriverNode) AddNodeInNodeDiskAvailabilityMap(ctx context.Context, nodeName string, nodeLabels map[string]string) {
+	capacity, _ := azureutils.GetNodeRemainingDiskCount(ctx, r.SharedState.cachedClient, nodeName)
+	var count atomic.Int32
+	count.Store(int32(capacity))
+	r.SharedState.nodeDiskAvailabilityMap.Store(nodeName, &count)
```

Comment:
Instead of storing unconditionally, it would be safer to only store the maximum capacity if an entry for the node doesn't exist.

Reply:
Done.